valhalla/baldr/traffictile.h

```diff
@@ -89,83 +67,25 @@ static_assert(sizeof(Speed) == sizeof(uint16_t),
  *
  * TileHeader (24 bytes)
  * n x Speed entries (n x 2 bytes)
- * IncidentsHeader (8 bytes)
- * m x Incidents entries (buffer 0) (m x 8 bytes)
- * m x Incidents entries (buffer 1) (m x 8 bytes)
- *
- * The Incidents are sparse - the TileHeader->incident_buffer_size specifies how
- * much space is available in the tile.  The IncidentsHeader fields specify where
- * incidents begin/end in this buffer.  This allows for new incident lists to be
- * placed in the buffer, then the begin/end updated to atomicly move readers to
- * a new location.
  */
 #ifndef C_ONLY_INTERFACE
+namespace {
+static constexpr volatile Speed INVALID_SPEED{0, 0, 0, 15, 0};
+}
 class Tile {
 public:
   Tile(char* tile_ptr)
-      : header{reinterpret_cast<TileHeader*>(tile_ptr)}, speeds{reinterpret_cast<volatile Speed*>(
-                                                             tile_ptr + sizeof(TileHeader))},
-        incident_count_0{reinterpret_cast<volatile uint32_t*>(
-            tile_ptr + sizeof(TileHeader) +
-            sizeof(Speed) * (tile_ptr == nullptr ? 0 : header->directed_edge_count))},
-        incident_count_1{reinterpret_cast<volatile uint32_t*>(
-            tile_ptr + sizeof(TileHeader) +
-            sizeof(Speed) * (tile_ptr == nullptr ? 0 : header->directed_edge_count) +
-            sizeof(*incident_count_0))},
-        incidents_0{reinterpret_cast<volatile Incident*>(
-            tile_ptr + sizeof(TileHeader) +
-            sizeof(Speed) * (tile_ptr == nullptr ? 0 : header->directed_edge_count) +
-            sizeof(*incident_count_0) * 2)},
-        incidents_1{reinterpret_cast<volatile Incident*>(
-            tile_ptr + sizeof(TileHeader) +
-            sizeof(Speed) * (tile_ptr == nullptr ? 0 : header->directed_edge_count) +
-            sizeof(*incident_count_0) * 2 +
-            (tile_ptr == nullptr ? 0 : header->incident_buffer_size) * sizeof(Incident))} {
+      : header{reinterpret_cast<volatile TileHeader*>(tile_ptr)},
+        speeds{reinterpret_cast<volatile Speed*>(tile_ptr + sizeof(TileHeader))} {
   }
 
-  Tile(const Tile& other) = default;
-  Tile& operator=(const Tile& other) = default;
-
-  const Speed getTrafficForDirectedEdge(const uint32_t directed_edge_offset) const {
+  const volatile Speed& getTrafficForDirectedEdge(const uint32_t directed_edge_offset) const {
     if (header == nullptr)
-      return {};
+      return INVALID_SPEED;
```

Comment:
haha this is exactly what i did in my attempt earlier today to change this locally :smile: